packages/apollo-engine-reporting/src/plugin.ts

```diff
@@ -66,6 +66,14 @@ export const plugin = <TContext>(
       schema,
       request: { http, variables },
     }) {
+      // If the options are false don't do any metrics timing.
+      if (options.traceReporting === false) {
+        // If we are running in gateway mode don't send the ftv1 trace
+        metrics.captureTraces = false;
```

Comment:
This is pretty confusing.  After looking for a while I see that this is related to the line
```
      reporting: !!this.engineReportingAgent,
```
in ApolloServer.graphQLServerOptions. The only thing that reads captureTraces is the Gateway executor, which doesn't happen until the execution phase (which is after this point).

So can we just totally eliminate `GraphQLServerOptions.reporting` (a flag I introduced purely in order to set `metrics.captureTraces`), initialize captureTraces to false in `runHttpQuery`.`buildRequestContext`, and make this function right here be the thing responsible for setting it to true in the first place? I hope there are no compatibility concerns involved in my hacky `reporting` flag that literally is just intended for this single purpose.

Reply:
Is it a breaking change to get rid of the `reporting` field from the `ApolloServer` config. I changed it to `captureTraces` by default is false.

Reply:
I'd noticed this, but it would be a breaking change as far as I could tell.

Reply:
So re `GraphQLServerOptions.reporting` / `GraphQLRequestMetrics.captureTraces`, that hacky pair of booleans I added in https://github.com/apollographql/apollo-server/pull/2900 with no docs other than a brief allusion in the PR description of “Thread some data through between reporting and gateway modules”…

I’m down to believe that literally deleting the `reporting` field is a bad idea for compat reasons, but perhaps we can at least make some subtler changes to make them a bit less hacky, esp given that in a world where we are actually changing the semantics around trace reporting and reporting to be less of a fixed boolean it makes sense that they may change?

Specifically, my thought is:
- continue to set `reporting `to `!!this.engineReportingAgent`, but document it (for the first time!) as "this is true if there’s an agent turned on, which does not necessarily mean that traces are being sent, oh and it’s deprecated"
- don't actually read `reporting` anywhere
- move the initialization of metrics.captureTraces into EngineReportingAgent

The effect, yes, is to make it so that `metrics.captureTraces` will be undefined in earlier phases of the request pipeline, instead of being set to true or false earlier… but I would argue that that’s actually an improvement, because if we actually are making “will traces be captured” into something more dynamic, then whether or not it is true really is undefined earlier in the pipeline, in a literal sense.

And if it turns out that anyone other than the gateway executor cares about the value of the `captureTraces` field and somebody complains, ok, I guess we could add the earlier initialization back, but the truth of the matter is that whatever plugin actually wants to know that will probably want to change anyway to deal with the fact that the calculation happens later!

I just think that I found this particular change very very confusing and I was the person who introduced it (without comments etc :) ). If all writes to captureTraces were within this file, it would be a lot easier to follow.

I think @abernix certainly has his finger on the pulse of how our open source software is used more than I do, but my instincts would tell me it’s pretty likely that zero third-party code looks at either of these booleans.  And certainly I’ve always regarded GraphQLRequestMetrics as purely for the sake of integrating Graph Manager reporting with other modules — in fact, the entire reason it’s its own sub-object of GraphQLRequestContext instead of part of the context is to isolate it and make its individual fields not part of the whole WithRequired chain, if I recall correctly. So I really don’t think we need to commit to the precise phase of the pipeline that fields on `GraphQLRequestMetrics` is set...